test/PrometheusPlugin.test.js

```diff
@@ -0,0 +1,76 @@
+const PrometheusPlugin = require('../lib/index.js'),
+  ContextMock = require('./mocks/context.mock'),
+  RequestMock = require('./mocks/request.mock'),
+  ConfigurationMock = require('./mocks/config.mock'),
+  Prometheus = require('prom-client'),
+  sinon = require('sinon');
+
+describe('PrometheusPlugin', () => {
+  let contextMock, plugin, configuration, request;
+
+  beforeEach(() => {
+    configuration = new ConfigurationMock();
+    contextMock = new ContextMock();
+    plugin = new PrometheusPlugin({}, contextMock);
+    request = new RequestMock();
+    Prometheus.register.clear(); // Clear registry object which is global
+    plugin.init(configuration, contextMock);
+    plugin.gateway.pushAdd = sinon.stub();
+  });
+
+  describe('#init', () => {
+    it('should create a PrometheusPlugin with correctly initialized metrics and hooks', () => {
+      plugin.gateway.should.be.an.instanceOf(Prometheus.Pushgateway);
```

Comment:
Nitpicking/optional: Usually we use the other syntax with `should`: `should(plugin.gateway).be.instanceOf(...)`